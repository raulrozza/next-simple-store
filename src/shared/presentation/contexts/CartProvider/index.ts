export { default as useAddToCart } from './hooks/useAddToCart';
