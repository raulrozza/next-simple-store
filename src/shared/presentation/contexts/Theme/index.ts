export { default as useTheme } from './hooks/useTheme';
export { default as Theme } from './implementations/StyledComponentTheme';
