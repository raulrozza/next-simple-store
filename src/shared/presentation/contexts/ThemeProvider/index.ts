export { default as useTheme } from './hooks/useTheme';
export { default as ThemeProvider } from './implementations/StyledComponentTheme';
