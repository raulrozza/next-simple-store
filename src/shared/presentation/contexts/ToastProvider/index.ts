export { useToastProvider } from './context';
export { default as ToastProvider } from './implementations/ReactToastifyProvider';
