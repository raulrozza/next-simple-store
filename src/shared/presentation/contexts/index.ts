export * from './CartProvider';
export * from './ThemeProvider';
