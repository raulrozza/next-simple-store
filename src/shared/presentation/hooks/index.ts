import { trpc } from '@/shared/infra/http/trpc';

const { useQuery, useMutation } = trpc;

export { useQuery, useMutation };
