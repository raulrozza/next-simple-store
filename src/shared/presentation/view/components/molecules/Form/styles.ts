import { Form } from 'formik';
import styled from 'styled-components';

export const StyledForm = styled(Form)`
    padding: ${({ theme }) => theme.layout.spacing(1)};
`;
