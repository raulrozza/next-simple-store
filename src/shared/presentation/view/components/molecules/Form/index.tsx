import { FC, Fragment } from 'react';

import { IField } from '@/shared/presentation/entities/Field';
import {
    Button,
    FieldFactory,
    Spacing,
} from '@/shared/presentation/view/components/atoms';

import { StyledForm } from './styles';

interface FormProps {
    fields: IField[];
    button: {
        text: string;
    };
}

const Form: FC<FormProps> = ({ fields, button }) => (
    <StyledForm>
        {fields.map(field => (
            <Fragment key={field.name}>
                <FieldFactory {...field} />

                <Spacing size={2} />
            </Fragment>
        ))}

        <Button type="submit" variant="secondary">
            {button.text}
        </Button>
    </StyledForm>
);

export default Form;
