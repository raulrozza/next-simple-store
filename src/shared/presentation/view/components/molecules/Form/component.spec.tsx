import { render, screen } from '@testing-library/react';
import { Formik } from 'formik';

import { ThemeProvider } from '@/shared/presentation/contexts';

import Form from './index';

describe('Form', () => {
    it('should render the Form correctly', () => {
        const view = render(
            <Formik
                initialValues={{
                    input: 'my input',
                }}
                onSubmit={jest.fn()}
            >
                <Form
                    fields={[
                        {
                            name: 'input',
                            type: 'text',
                            placeholder: 'my input',
                        },
                    ]}
                    button={{
                        text: 'Confirm',
                    }}
                />
            </Formik>,
            {
                wrapper: ThemeProvider,
            },
        );

        expect(screen.getByPlaceholderText('my input')).toBeInTheDocument();
        expect(screen.getByText('Confirm')).toBeInTheDocument();
        expect(view).toMatchSnapshot();
    });
});
