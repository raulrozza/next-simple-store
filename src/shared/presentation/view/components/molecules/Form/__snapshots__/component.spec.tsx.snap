// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render the Form correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c3 {
  padding: 0.5rem;
  border-radius: 0.5rem;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.25rem;
  background-color: #754A56;
  color: #FFFFFF;
}

.c3:hover {
  background-color: #845864;
}

.c1 {
  background-color: #FFFFFF;
  border: 2px solid #E6EAEA;
  outline: none;
  border-radius: 0.25rem;
  width: 100%;
  padding: 0.5rem;
  -webkit-transition: outline 0.2s,border 0.2s;
  transition: outline 0.2s,border 0.2s;
}

.c1:focus {
  border-color: #8CCFDE;
}

.c2 {
  margin-top: 0.5rem;
}

.c0 {
  padding: 0.25rem;
}

<body>
    <div>
      <form
        action="#"
        class="c0"
      >
        <div>
          <input
            class="c1"
            name="input"
            placeholder="my input"
            value="my input"
          />
        </div>
        <div
          class="c2"
          size="2"
        />
        <button
          class="c3"
          type="submit"
        >
          Confirm
        </button>
      </form>
    </div>
  </body>,
  "container": .c3 {
  padding: 0.5rem;
  border-radius: 0.5rem;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.25rem;
  background-color: #754A56;
  color: #FFFFFF;
}

.c3:hover {
  background-color: #845864;
}

.c1 {
  background-color: #FFFFFF;
  border: 2px solid #E6EAEA;
  outline: none;
  border-radius: 0.25rem;
  width: 100%;
  padding: 0.5rem;
  -webkit-transition: outline 0.2s,border 0.2s;
  transition: outline 0.2s,border 0.2s;
}

.c1:focus {
  border-color: #8CCFDE;
}

.c2 {
  margin-top: 0.5rem;
}

.c0 {
  padding: 0.25rem;
}

<div>
    <form
      action="#"
      class="c0"
    >
      <div>
        <input
          class="c1"
          name="input"
          placeholder="my input"
          value="my input"
        />
      </div>
      <div
        class="c2"
        size="2"
      />
      <button
        class="c3"
        type="submit"
      >
        Confirm
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
